admission_control_overload_test.py

```diff
@@ -0,0 +1,91 @@
+            time.sleep(10)
+
+        # run a read workload
+        stress_queue = []
+        stress_res = []
+        stress_queue.append(self.run_stress_thread(stress_cmd=stress_base_cmd, stress_num=job_num,
+                                                   stats_aggregate_cmds=False))
+
+        start_time = time.time()
+        max_transport_requests_memory_available = 0  # pylint: disable=invalid-name
+        is_admission_control_triggered = False
+        while stress_queue:
+            stress_res.append(stress_queue.pop(0))
+            while not all(future.done() for future in stress_res[-1].results_futures):
+                now = time.time()
+                transport_res = prometheus.query(transport_requests_memory_available, start_time, now)
+                node_procs_res = prometheus.query(node_procs_blocked, start_time, now)
+
+                sum_values = 0
+                for shard in transport_res:
+                    sum_values += int(shard['values'][0][1])
+                max_transport_requests_memory_available = max(  # pylint: disable=invalid-name
+                    sum_values, max_transport_requests_memory_available)
+                if sum_values < max_transport_requests_memory_available * 0.3:
+                    max_transport_requests_memory_available = sum_values  # pylint: disable=invalid-name
+                    for node in node_procs_res:
+                        if node['values'][0][1] > 0:
+                            self.log.info('Admission control was triggered')
+                            is_admission_control_triggered = True
+                start_time = now
+                time.sleep(10)
+
+        results = []
+        for stress in stress_res:
+            results.append(self.get_stress_results(stress))
+        # results = self.verify_stress_thread(cs_thread_pool=stress_queue)
+
+        self.verify_no_drops_and_errors(starting_from=self.start_time)
+        # TODO: raise exception if is_admission_control_triggered is false
+        if not is_admission_control_triggered:
+            raise Exception('Admission Control wasn\'t triggered')
+
+    def test_read_admission_control_under_load(self):  # pylint: disable=invalid-name
+        """
+        Test steps:
+        1. Run a write workload as a preparation
+        2. Run a read workload
+        """
+
+        base_cmd_prepare = self.params.get('prepare_write_cmd')
+        base_cmd_r = self.params.get('stress_cmd_r')
+
+        self.run_load_and_check_for_admission_control(base_cmd_prepare, base_cmd_r)
+
+    def test_write_admission_control_under_load(self):  # pylint: disable=invalid-name
+        """
+        Test steps:
+        1. Run a write workload as a preparation
+        """
+        # FIXME: might have to increase the write load, as it didn't trigger the admission control as in the read
+        base_cmd_w = self.params.get('stress_cmd_w')
+
+        self.run_load_and_check_for_admission_control(None, base_cmd_w, job_num=16)
```

Comment:
if don't have scp issue with 8 jobs, but still not triggering the admission control add more loaders.

Reply:
done a workaround on the scp remoter command, and it is not an issue anymore (until refactor)